data/service/api/v1/datasets_data_create.go

```diff
@@ -136,28 +137,26 @@ func DataSetsDataCreate(dataServiceContext dataService.Context) {
 // CollectProvenanceInfo from a request and its auth details.
 //
 // All work is done as a best effort right now.
-func CollectProvenanceInfo(ctx context.Context, req *rest.Request, details request.AuthDetails) *data.Provenance {
+func CollectProvenanceInfo(ctx context.Context, req *rest.Request, authDetails request.AuthDetails) *data.Provenance {
 	lgr := log.LoggerFromContext(ctx)
 	provenance := &data.Provenance{}
 
-	token := req.Header.Get(auth.TidepoolSessionTokenHeaderKey)
-	if token == "" {
-		token = req.Header.Get(auth.TidepoolAuthorizationHeaderKey)
-		if token != "" {
-			if strings.EqualFold(token[:len("bearer ")], "bearer ") {
-				token = token[len("bearer "):]
-			}
+	token := authDetails.Token()
+	if token != "" && strings.Contains(strings.ToLower(token), "bearer") {
+		if strings.EqualFold(token[:len("bearer ")], "bearer ") {
+			token = token[len("bearer "):]
 		}
 	}
```

Comment:
Perhaps simpler?
```suggestion
    if strings.BeginsWith(strings.ToLower(token), "bearer ") {
		token = token[len("bearer "):]
	}
```

Reply:
Yes, thanks! I'll just double check the logic works out okay, then pull this in.

Reply:
Yes, thanks.